### PR TITLE
eduke32: migrate to pkgs/by-name, format with nixfmt-rfc-style, 0-unstable-2024-02-17 -> 0-unstable-2024-07-26, add ion fury and fix wrappers

### PR DESCRIPTION
**Pull request by qubitnano**
## Description of changes

- Migrate to pkgs/by-name
- Format with nixfmt-rfc-style
- Update to latest hash
- Add ion fury wrapper and desktop icon
- Rework wrapper logic. Set GRP file per wrapper so when it traverses looking for games it will force to load the correct one

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by mikroskeem** [comment]
Can you remove me from maintainers list also?

**Review by mikroskeem — APPROVED** [review]

**Comment by keenanweaver** [comment]
Result of `nixpkgs-review pr 311494` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>eduke32</li>
  </ul>
</details>

**Review by keenanweaver — APPROVED** [review]
All GRPs & addons playing as expected.

**Comment by qubitnano** [comment]
I'd like to help maintain this package :)

**Comment by keenanweaver** [comment]
Result of `nixpkgs-review pr 311494` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>eduke32</li>
  </ul>
</details>

**Review by keenanweaver — APPROVED** [review]
image

Working great, especially with nuked-sc55!

File: pkgs/games/eduke32/default.nix [hunk 1]
```diff
@@ -112,48 +124,71 @@ in stdenv.mkDerivation (finalAttrs: {
...
+    ''
+    + lib.optionalString stdenv.isLinux ''
+      makeWrapper $out/bin/eduke32 $out/bin/${wrapper} \
+        --set-default EDUKE32_DATA_DIR /var/lib/games/eduke32 \
+        --add-flags '-j "$EDUKE32_DATA_DIR"' \
+        --add-flags '-gamegrp DUKE3D.GRP'
+      makeWrapper $out/bin/voidsw $out/bin/${swWrapper} \
+        --set-default EDUKE32_DATA_DIR /var/lib/games/eduke32 \
+        --add-flags '-j"$EDUKE32_DATA_DIR"' \
+        --add-flags '-gSW.GRP'
```
Comment (mikroskeem) [inline pkgs/games/eduke32/default.nix:154]:
```suggestion
        --add-flags '-j "$EDUKE32_DATA_DIR"' \
        --add-flags '-g SW.GRP'
```

Style inconsistency compared to other wrappers

**Review by mikroskeem — CHANGES_REQUESTED** [review]

File: pkgs/games/eduke32/default.nix (hunk 1 above)
Comment (qubitnano) [reply pkgs/games/eduke32/default.nix]: `voidsw` requires `-j` without the space, but `eduke32` also works without the space, so I made those consistent.

Also `voidsw` only looks for SW.GRP so I removed that line.

**Review by mikroskeem — APPROVED** [review]

**Comment by keenanweaver** [comment]
Result of `nixpkgs-review pr 311494` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>eduke32</li>
  </ul>
</details>

**Review by keenanweaver — APPROVED** [review]

**Comment by keenanweaver** [comment]
Result of `nixpkgs-review pr 311494` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>eduke32</li>
  </ul>
</details>

**Review by keenanweaver — APPROVED** [review]

**Pull request merged**